src/WebJobs.Extensions.DurableTask/Listener/TaskOrchestrationShim.cs

```diff
@@ -133,7 +133,20 @@ public override async Task<string> Execute(OrchestrationContext innerContext, st
             {
                 if (orchestratorInfo.IsOutOfProc)
                 {
-                    var jObj = returnValue as JObject;
+                    JObject jObj = null;
+
+                    if (returnValue is JObject)
+                    {
+                        jObj = returnValue as JObject;
+                    }
+                    else
+                    {
+                        if (returnValue is string)

```

Comment:
Consider the following implementation as a means of reducing the lines of code (without the comments):

```csharp
// same as before
var jObj = returnValue as JObject;

// new code
if (jObj == null && returnValue is string jsonText)
{
    jObj = JObject.Parse(jsonText);
}
```

Reply:
Thank you very much @cgillum !

Your optimization is great!

Please let me know if we need to change something else.

Thank you!